src/usr/xio/xio_mem.c

```diff
@@ -520,3 +520,19 @@ void xio_numa_free_ptr(void *ptr)
 		   */
 		xio_numa_free(real_ptr, real_size);
 }
+
+/*---------------------------------------------------------------------------*/
+/* xio_mem_alloc_ex							     */
+/*---------------------------------------------------------------------------*/
+inline int xio_mem_alloc_ex(size_t length, struct xio_reg_mem *reg_mem,
+			    struct xio_reg_mem_opt *opt)
+{
+	if (length == 0 || !reg_mem) {
```

Comment:
Since this check also needs to be done in xio_mem_alloc(), please do something like:

```
  if (opt->register_mem == 0) {
          if (length == 0 || !reg_mem) {
                  ...
                  return -1;
          }
          return xio_mem_alloc_no_dev(...);
  }

  return xio_mem_alloc(...);
```